manifests/repos.pp

```diff
@@ -22,19 +27,42 @@
       $package_key   = $docker::package_key
       $package_repos = $docker::package_repos
 
+      if ( $facts['os']['distro']['id'] == 'Debian' and versioncmp($facts['os']['distro']['release']['major'],'10') >= 0 ) or ( $facts['os']['distro']['id'] == 'Ubuntu' and versioncmp($facts['os']['distro']['release']['major'],'22') >= 0 ) { # lint:ignore:140chars
+        # fix deprecated apt-key warnings
+        file { $keyring:
+          ensure  => file,
+          content => binary_file("${module_name}/etc/apt/keyrings/docker.gpg"),
```

Comment:
I'm a bit unsure about this. This adds some technical debt because now someone needs to check the gpg file on a regular basis in this repository and update it. Usually we fetch those keys directly from upstream.  Does docker offer it somewhere on their website where it could be downloaded? maybe make the content and source attributes of the file resource parameterized.

Reply:
https://download.docker.com/linux/ubuntu/gpg
https://download.docker.com/linux/debian/gpg

I will later take a look how to implement it.  
But I agree with you, that would cause a technical debt.  
We just have to make sure not to use any keyservers (which is no longer recommended).

Reply:
Added via a1b7602b4c27288ce9c64269fb5eb3f3fcffb725 .
Also added function to force update.  

Since in my opinion download.docker.com should be the only 'source-of-truth' I didn't parameterized the download-URL.